src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,123 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+import duke.common.DukeException;
+
+/**
+ * A task specified by the user to Duke.
+ *
+ * @author Rama Aryasuta Pangestu
+ */
+public abstract class Task {
+    protected static final String ENCODING_SEPARATOR = " | ";
+    protected static final String DECODING_SEPARATOR = " \\| ";

```

Comment:
Have you considered naming them `SEPARATOR_ENCODING` and `SEPARATOR_DECODING` instead?